pandas/io/parsers/arrow_parser_wrapper.py

```diff
@@ -158,6 +160,19 @@ def read(self) -> DataFrame:
             parse_options=pyarrow_csv.ParseOptions(**self.parse_options),
             convert_options=pyarrow_csv.ConvertOptions(**self.convert_options),
         )
+
+        # Convert all pa.null() cols -> float64
+        # TODO: There has to be a better way... right?
```

Comment:
What I'm looking for here is something like a combination of ``select_dtypes`` + ``astype``.

I couldn't find anything in the pyarrow docs, but maybe I'm just bad at Googling.

Sadly the types_mapper doesn't work, since you can only feed that EA types.

Reply:
Maybe @jorisvandenbossche knows a better way?

Reply:
Does Table.select not work?

Reply:
I don't think there is an easier way. We currently don't have something like select_dtypes, so what you do (iterating over the schema, checking the type, and if needed adapt the new schema) seems the best option.